### PR TITLE
Trivia: Use Room#subGame for Mastermind

### PR DESCRIPTION
**Pull request by AnnikaCodes**
@Zarel and @sparkychildcharlie, I think this is what you were proposing? It makes `Room#mastermindGame` track the Mastermind game, and uses `Room#game` for the rounds and finals. I also made `RoomGamePlayer`s independent of the game since all the state we really need is the room ID.

### CODE REVIEW

File: server/rooms.ts [hunk 1]
```diff
@@ -179,6 +180,7 @@ export abstract class BasicRoom {
 	scavLeaderboard: AnyObject;
 	giveaway?: QuestionGiveaway | LotteryGiveaway | null;
 	gtsga?: GTSGiveaway | null;
+	mastermindGame?: Mastermind | null;
```
Comment (Zarel) [inline server/rooms.ts:183]: Is there any reason why this needs to be directly in room, instead of just `room.game.mastermindGame` or something?

File: server/room-game.ts [hunk 2]
```diff
@@ -18,7 +18,7 @@
 // globally Rooms.RoomGamePlayer
 export class RoomGamePlayer {
 	readonly num: number;
-	readonly game: RoomGame;
+	readonly roomid: RoomID;
```
Comment (Zarel) [inline server/room-game.ts:21]: Why was this changed?

File: server/rooms.ts (hunk 1 above)
Comment (AnnikaCodes) [reply server/rooms.ts:183]: Well, `room.game` represents a round of Mastermind here, so `mastermindGame` needs to exist while `room.game`  is null.

File: server/room-game.ts (hunk 2 above)
Comment (AnnikaCodes) [reply server/room-game.ts:21]: I changed this so that I could use `RoomGamePlayer`s to track Mastermind players without technically having a `RoomGame` in progress.

File: server/rooms.ts (hunk 1 above)
Comment (Zarel) [reply server/rooms.ts:183]: Would it be very hard to do it the other way around? Where `room.game` is the entire Mastermind game, and `room.game.round` is a round of Mastermind?

Comment (Zarel) [reply server/rooms.ts:183]: That would also fix your `RoomGamePlayer` issue.

Comment (AnnikaCodes) [reply server/rooms.ts:183]: That would be _possible_ but probably overcomplicated, since Mastermind rounds are just slightly modified Trivia games, and we still want Trivia to extend RoomGame.

Comment (Zarel) [reply server/rooms.ts:183]: I... would still massively prefer that to the weirdness going on in this PR which the RoomGame system was never designed to support.

If this is a frequent problem, I think the correct way to do this would be to spec out a `.subGame` property on roomgames which can take `RoomGame` instances.

Comment (Zarel) [reply server/rooms.ts:183]: Either way, a simple solution is just not using RoomGame the way it was designed to be used and would need to be carefully audited and end up overcomplicated no matter what.

**Comment by AnnikaCodes** [comment]
I'm gonna draft this until I make `RoomGame`s work somewhere other than `room.game`.

File: server/rooms.ts (hunk 1 above)
Comment (AnnikaCodes) [reply server/rooms.ts]: This uses `Room#subGame` now.

File: server/room-game.ts (hunk 2 above)
Comment (AnnikaCodes) [reply server/room-game.ts]: This is no longer necessary.

**Comment by Zarel** [comment]
Thanks! Hopefully this is all it needed.
